geopandas/tests/test_crs.py

```diff
@@ -435,6 +437,86 @@ def test_read_file(self, nybb_filename):
         assert df.geometry.crs == pyproj.CRS(2263)
         assert df.geometry.values.crs == pyproj.CRS(2263)
 
+    def test_read_file_warning_messages(self):
```

Comment:
we should parametrise with engine over fiona and pyogrio. For that reason it may be easier to move this test into `test_file.py` to reuse the `engine` fixture, and then also add `@pytest.mark.skipif(not HAS_PYPROJ, reason="pyproj not installed")` so that this test will only run with pyproj (we now support having both no pyproj and neither io engine, and by putting this test in test_crs.py as written it would require pyogrio or fiona to be installed to run test_crs.